stdlib/public/stubs/LibcShims.cpp

```diff
@@ -308,3 +318,72 @@ swift::_stdlib_cxx11_mt19937_uniform(__swift_uint32_t upper_bound) {
   std::uniform_int_distribution<__swift_uint32_t> RandomUniform(0, upper_bound);
   return RandomUniform(getGlobalMT19937());
 }
+
+// _stdlib_random
+//
+// Should the implementation of this function add a new platform/change for a
+// platform, make sure to also update the stdlib documentation regarding
+// platform implementation of this function.
+// This can be found at: stdlib/public/core/Random.swift
+
+#if defined(_WIN32) && !defined(__CYGWIN__)
+#error TODO: Test _stdlib_random on Windows
+
+SWIFT_RUNTIME_STDLIB_INTERNAL
+void swift::_stdlib_random(void *buf, __swift_size_t nbytes) {
+  NTSTATUS status = BCryptGenRandom(nullptr,
+                                    static_cast<PUCHAR>(buf),
+                                    static_cast<ULONG>(nbytes),
+                                    BCRYPT_USE_SYSTEM_PREFERRED_RNG);
+  if (!BCRYPT_SUCCESS(status)) {
+    fatalError(0, "Fatal error: 0x%.8X in '%s'\n", status, __func__);
+  }
+}
+
+#else
+
+#undef  WHILE_EINTR
+#define WHILE_EINTR(expression) ({                                             \
+  decltype(expression) result = -1;                                            \
+  do { result = (expression); } while (result == -1 && errno == EINTR);        \
+  result;                                                                      \
+})
+
+SWIFT_RUNTIME_STDLIB_INTERNAL
+void swift::_stdlib_random(void *buf, __swift_size_t nbytes) {
+  while (nbytes > 0) {
+    __swift_ssize_t actual_nbytes = -1;
+#if defined(GRND_RANDOM)
+    static const bool getrandom_available =
+      !(getrandom(nullptr, 0, 0) == -1 && errno == ENOSYS);
+    if (getrandom_available) {
+      actual_nbytes = WHILE_EINTR(getrandom(buf, nbytes, 0));
+    }
+#elif defined(__Fuchsia__)
+    __swift_size_t getentropy_nbytes = std::min(nbytes, __swift_size_t{256});
+    if (0 == getentropy(buf, getentropy_nbytes)) {
+      actual_nbytes = getentropy_nbytes;
+    }
+#elif defined(__APPLE__)
+    arc4random_buf(buf, nbytes);
+    actual_nbytes = nbytes;
+#endif
+    if (actual_nbytes == -1) {
+      static const int fd =
```

Comment:
I'm not sure if this code should apply to `arc4random` on Darwin platforms, which claims it never returns an error code.

Reply:
This does not apply to `arc4random` on Darwin platforms and won't be executed during runtime. I can separate Darwin's solution outside of this one if that clears any confusion.